### PR TITLE
Feat: New API `WorkflowRun.get_artifact()` and `WorkflowRun.get_artifact_serialized()` to get single task output

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
We lacked the ability to get single artifact based on invocation ID. This caused us to always fetch all the artifacts, which is problematic for workflows with lots of artifacts where each individual artifacts is memory-heavy 

# This PR's solution
Add new API to get single artifact - now only implemented on Ray Runtime

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

**Review by alexjuda — APPROVED** [review]

**Comment by alexjuda** [comment]
Took me a while to consider possible edge cases related to tasks with multiple outputs (`n_outputs=2`) but your unit tests and the `RuntimeInterface` docstring seem to cover it

File: src/orquestra/sdk/_base/abc.py [hunk 1]
```diff
@@ -116,6 +116,29 @@ def get_available_outputs(
...
+        Verify it beforehand to make sure if task failed/succeeded/is running.
+        You might get an exception
+
+        Args:
+            workflow_run_id: ID identifying the workflow run.
+            task_invocation_id: ID identifying single task run
+
+        Returns:
+            Whatever the task function returned, independent of the
+            ``@task(n_outputs=...)`` value.
```
Comment (alexjuda) [inline src/orquestra/sdk/_base/abc.py:138]: **suggestion**: Just for peace of mind, could you add an integration test that verifies that our Ray runtime implementation functions correctly with a task that has `n_outputs` > 1?

Comment (alexjuda) [reply src/orquestra/sdk/_base/abc.py:138]: By "function correctly" I mean it should be consistent with this docstring.
Given:
* a task def with `n_outputs=2`
* task returns a `"foo", "bar"`
Expect:
* `runtime.get_output(...)[0] == "foo"`
* `runtime.get_output(...)[1] == "bar"`

Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/abc.py:138]: rutnime.get_output[...] woudlnt work, as it returns JSON-encoded thing rather than tuple. But Ive added test to verify everythign works correctly
